src/compiler/c.rs

```diff
@@ -390,11 +402,38 @@ where
                 arg
             );
         }
+
+        let use_preprocessor_cache_mode = {
```

Comment:
for example, this change seems to be independent from nvcc

Reply:
It may seem that way, but unfortunately it is not.

The generated subcommands [are run](https://github.com/trxcllnt/sccache/blob/282ea920deb5c2baf4b236804a2e359518c04004/src/compiler/nvcc.rs#L1036) with `SCCACHE_DIRECT` set to false, because the preprocessed output is either impossible (or unnecessary) to cache, so attempting to cache it is just wasted cycles and messier logs.

It didn't seem like there was a good way to disable the preprocessor caching besires the envvar, but then the issue is that this envvar is only read on startup and not per file, so this change ensures it's read per file.

Reply:
sorry for the latency but maybe add a comment in the code too :)